core/src/main/java/org/wildfly/glow/GlowSession.java

```diff
@@ -828,6 +829,105 @@ private static void fixAddOns(ErrorIdentificationSession errorSession,
         Set<String> familyOfAddOnsComplete = new TreeSet<>();
         Map<String, Set<AddOn>> membersInFamily = new HashMap<>();
         Map<String, Set<AddOn>> defaultMembersInFamily = new HashMap<>();
+        // Handle inclusion mode
+        Map<String, Set<AddOn>> selectedAddOnsForFamily = new HashMap<>();
+        Map<String, Set<AddOn>> candidatesForFamily = new HashMap<>();
+        Set<AddOn> toBeRemoved = new HashSet<>();
+        for(AddOn ao : allEnabledAddOns) {
+            SelectionMode mode = mapping.getFamilySelectionMode().get(ao.getFamily());
+            // We have a selection mode for this family
+            // A selection means that such add-ons must be included by the user and not automatically discovered.
+            if (mode != null) {
+                Set<AddOn> selectedAddOns = selectedAddOnsForFamily.get(ao.getFamily());
+                if (selectedAddOns == null) {
+                    selectedAddOns = new HashSet<>();
+                    selectedAddOnsForFamily.put(ao.getFamily(), selectedAddOns);
+                }
```

Comment:
Suggestion: use computeIfAbsent.

I normally don't make that kind of code style comment, but this method is very very complex and can benefit from this kind of simplification.

Reply:
Thanks, fixed.